File: frontend/src/types/colaborador.ts
export interface Colaborador {
    id: string;

    nome: string;
    matricula: string;

    email: string | null;
    telefone: string | null;

    cargo: string;
    setor: string;
    cidade: string;
    dia_folga: string | null;

    data_admissao: string;

    status: string;

    foto_url: string | null;
    observacoes: string | null;

    gestor_id: string | null;
    gestor_nome: string | null;
}

File: backend/app/schemas/colaborador.py
from datetime import date
from uuid import UUID
from pydantic import BaseModel, ConfigDict
from app.models.status_colaborador import StatusColaborador

class ColaboradorBase(BaseModel):
    nome: str
    matricula: str
    email: str | None = None
    telefone: str | None = None
    cargo: str
    setor: str
    cidade: str
    data_admissao: date
    status: StatusColaborador = StatusColaborador.ATIVO
    foto_url: str | None = None
    observacoes: str | None = None
    gestor_id: UUID | None = None
    dia_folga: str | None = None

class ColaboradorCreate(ColaboradorBase):
    pass

class ColaboradorUpdate(BaseModel):
    nome: str | None = None
    matricula: str | None = None
    email: str | None = None
    telefone: str | None = None
    cargo: str | None = None
    setor: str | None = None
    cidade: str | None = None
    data_admissao: date | None = None
    status: StatusColaborador | None = None
    foto_url: str | None = None
    observacoes: str | None = None
    gestor_id: UUID | None = None
    dia_folga: str | None = None

class ColaboradorResponse(ColaboradorBase):
    id: UUID

    model_config = ConfigDict(
        from_attributes=True,
    )